python/DEVELOPMENT.md

```diff
@@ -67,15 +45,36 @@ drwxrwxr-x 12 wesm wesm 4096 Apr 15 19:19 arrow/
 drwxrwxr-x 12 wesm wesm 4096 Apr 15 19:19 parquet-cpp/
 ```
 
+We need to set a number of environment variables to let Arrow's build system
+know about our build toolchain:
+
+```
+export ARROW_BUILD_TYPE=release
+
+export BOOST_ROOT=$CONDA_PREFIX
```

Comment:
We could e.g. just try and search first for all dependencies in `$CMAKE_INSTALL_PREFIX`